tests/contrib/django/django_tests.py

```diff
@@ -242,6 +247,25 @@ def test_user_info_with_non_django_auth(self):
         event = self.elasticapm_client.events.pop(0)['errors'][0]
         assert event['context']['user'] == {}
 
+    @pytest.mark.skipif(django.VERSION < (1, 10),
+                        reason='MIDDLEWARE new in Django 1.10')
+    def test_user_info_with_non_django_auth_django_2(self):
+        with self.settings(INSTALLED_APPS=[
+            app for app in settings.INSTALLED_APPS
+            if app != 'django.contrib.auth'
+        ]) and self.settings(MIDDLEWARE_CLASSES=None, MIDDLEWARE=[
```

Comment:
except of the line `self.settings(MIDDLEWARE_CLASSES=[..]` this seems to be the exact same test for `django1` and `django2`. Since it seems that both of the tests will be in the code base for quite some time, what do you think about creating a common method for the test logic and just having the specific parts explicitly defined in the specific methods?

Reply:
That's what I did for most tests with the `middleware_setting` helper function -- it generates the correct settings dict on the fly for the current Django version. I couldn't do that here, because we _remove_ a specific middleware, and leave the rest as is. I could have written another helper function, but because we only use this pattern once or twice, it didn't feel like the right thing to do.